.github/actions/build/action.yml

```diff
@@ -0,0 +1,66 @@
+name: 'Build'
+description: 'Build nix'
+inputs:
+  # This is required
+  TARGET:
+    required: true
+
+  BUILD:
+    required: false
+    default: build
+   
+  CLIPPYFLAGS: 
+    required: false
+    default: -D warnings -A unknown-lints
+
+  RUSTFLAGS: 
+    required: false
+    default: -D warnings -A unknown-lints
+
+  RUSTDOCFLAGS: 
+    required: false
+    default: -D warnings
+
+  TOOL: 
+    description: 'Tool used to involve the BUILD command, can be cargo or cross'
+    required: false
+    default: cargo
+
+  ZFLAGS:
+    required: false
+    default: 
+
+  NOHACK:
+    description: "whether to run cargo hack"
+    required: false
+    default: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: debug info
+      shell: bash
+      run: |
+        ${{ inputs.TOOL }} -Vv
+        rustc -Vv
+
+    - name: build
```

Comment:
For these environment variables, it might be better to add a dedicated step for it:

```yaml
- name: Setup Rust env
  run: |
    echo "RUSTFLAGS= ${{ inputs.RUSTFLAGS }}" >> $GITHUB_ENV
    echo "RUSTDOCFLAGS= ${{ inputs.RUSTDOCFLAGS }}" >> $GITHUB_ENV
```

Reply:
Agreed, this is better as every subsequent step needs those env variables